File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
#include <sstream>
#include <string>
#include <vector>
using namespace std;
class Solution {
public:
    string reverseWords(string s) {
        stringstream ss(s);
        string token;
        vector<string> tokens;
        string res;

        while (ss >>token){
            tokens.push_back(token);
        }

        for (int i=tokens.size()-1;i>=0;i--){
            res+=tokens[i];
            if (i>0){
                res+=" ";
            }
        }
        return res;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        res=""
        curr=""
        words=[]
        for i in s:
            if i!=" ":
                curr+=i
            else:
                if curr!="":
                    words.append(curr)
                    curr=""

        if curr:
                words.append(curr)

        return " ".join(words[::-1])